packages/launchpad/cypress/e2e/config-warning.cy.ts

```diff
@@ -98,7 +112,7 @@ describe('experimentalStudio', () => {
     cy.get('[data-cy="warning-alert"]').contains('Warning: Experimental Studio Removed')
   })
 
-  it('should not continually show experimentalStudio warning in the same project', () => {
+  it('should show experimentalStudio warning in the same project', () => {
```

Comment:
This test was added specifically to ensure warnings are clearable and they don't persist when switching testing types. See https://github.com/cypress-io/cypress/pull/21630.

I think we'll want to figure out what we need to hook into to maintain this functionality.

Reply:
Right, I looked into this a bit more. There's actually two kinds of warnings.

The edge case here is *global config* warnings. Eg consider

```ts
export default defineConfig({
  experimentalStudio: true,
  e2e: {
    baseUrl: 'http://this-does-not-exist:2222'
  }
})
```

Since we only execute `e2e` and `component` after selecting a testing type, and we clear those warnings when we change testing types, it's fine - the behavior is as expected.

The difference with `experimentalStudio` (which I'm calling a "global" config) is those warnings show up on the "selecting testing type" screen. So we need some kind of way to handle those. The majority (well, all?) of the deprecated global config result in an error (since we've got the migration UI, it's rare users will hit this):

<img width="1312" alt="image" src="https://user-images.githubusercontent.com/19196536/185260442-5b4b9444-a722-43b3-8ea8-5f90fe3b05aa.png">

I think `experiementalStudio` is the only *warning* (not error) that is global. 

Either way good catch - I'll look into some logic to cover this case.

Reply:
Done... https://github.com/cypress-io/cypress/pull/23339/commits/081cba8154c7785a3570abd09b3cf9d4198d4617 🙏

Reply:
I think the root of double source of truth is these function signatures...

https://github.com/cypress-io/cypress/blob/develop/packages/config/src/browser.ts#L78


```js
const validateNoBreakingOptions = (
  breakingCfgOptions: Readonly<BreakingOption[]>,
  cfg: any, 
  onWarning: ErrorHandler, 
  onErr: ErrorHandler, 
  testingType?: TestingType) => {
  // ...
}
```

We should not be passing `onWarning` and `onErr` from data-context to config. Instead, `validateNoBreakingOptions` should be `get BreakingOptions`, which would be dealt with in data-context:

```js
const { errors, warnings } = get BreakingOptions(config, testingType)

if (warnings) {
  onWarning(warnings)
)

if (errors) {
  onError(errors)
}
```

To better represent the separation of responsibilities. Anyway, this is a big lift, but good to know how this initial spreading of truth was introduced and a nice drive-by refactor if we are in this part of the code base again in the future. I really think config should *not* have side effects.